pkg/sdk/request.go

```diff
@@ -64,79 +80,100 @@ func (m *msgBrokerSvc) RegisterSub(method string, callbackUrl string, publicKey
 	return nil
 }
 
+// RegisterPub registers a publisher with the message broker service using the provided channel. It returns the ID of the registered publisher and an error if any occurred.
 func (m *msgBrokerSvc) RegisterPub(channel string) (string, error) {
-	var response model.Response
+	// Create a new publisher object with the given channel
 	pub := model.Publisher{Channel: channel}
+	// Convert the publisher object to a JSON byte array
 	reqBody, err := json.Marshal(pub)
 	if err != nil {
 		return "", err
 	}
+	// Create a new HTTP client with a 2 second timeout
 	client := http.Client{
 		Timeout: 2 * time.Second,
 	}
+	// Send a POST request to the message broker's /register/publisher endpoint with the JSON request body
 	r, err := client.Post(m.msgbrokerUrl+"/register/publisher", "application/json", bytes.NewBuffer(reqBody))
 	if err != nil {
 		return "", err
 	}
+	// If the HTTP response status code is not in the 200-299 range, return an error
 	if r.StatusCode < 200 || r.StatusCode > 299 {
 		return "", fmt.Errorf("non success status code received : %v", r.StatusCode)
 	}
+	// Read the response body into a byte array
 	body, err := ioutil.ReadAll(r.Body)
 	if err != nil {
 		return "", err
 	}
+	// Unmarshal the JSON response body into a Response object
+	var response model.Response
 	err = json.Unmarshal(body, &response)
 	if err != nil {
 		return "", err
 	}
+	// Extract the data field from the Response object and convert it to a map[string]interface{}
 	data, ok := response.Data.(map[string]interface{})
 	if !ok {
 		return "", fmt.Errorf("unexpected response")
 	}
+	// Extract the ID field from the map and convert it to a string
 	id, ok := data["id"].(string)
 	if !ok || len(id) == 0 {
 		return "", fmt.Errorf("id not found")
 	}
+	// Return the ID of the newly registered publisher
 	return id, nil
 }
 
+// PushMsg sends a message to the message broker service
 func (m *msgBrokerSvc) PushMsg(msg string, uuid string, channel string) error {
+	// create a new message update with the message, publisher ID and channel
 	var update = model.Updates{
 		Update: msg,
 		Publisher: model.Publisher{
 			Id:      uuid,
 			Channel: channel,
 		},
 	}
+	// marshal the update to JSON
 	reqBody, err := json.Marshal(update)
 	if err != nil {
 		return err
 	}
+	// send a POST request to the message broker service's /publish endpoint with the update in the request body
 	client := http.Client{
 		Timeout: 2 * time.Second,
 	}
 	r, err := client.Post(m.msgbrokerUrl+"/publish", "application/json", bytes.NewBuffer(reqBody))
 	if err != nil {
 		return err
 	}
+	// check that the response status code indicates success
 	if r.StatusCode < 200 || r.StatusCode > 299 {
 		return fmt.Errorf("non success status code received : %v", r.StatusCode)
 	}
 	return nil
 }
 
+// ExtractMsg returns a function that can extract a message from a ReadCloser
```

Comment:
also mention the purpose of the key and what happens if not provided...encrypted/unencrypted messages

Reply:
done